### DISCUSSION TITLE
Python的程式執行方式與其它程式有不同嗎？

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#350] [Q&A]
老師在上課說因為Python是腳本語言(script language)，因此程式會由上至下一行一行執行。既然老師特別強調Python是腳本語言，那應該代表如C、C++、java、javascript、R與其它所有的程式語言都不是腳本語言。不是腳本語言的話，程式執行的方式應該也不是由上至下一行一行執行。可是，怎麼看起來每一種程式執行(跑)的方式，都是由上至下一行一行執行？
🇶🇦

**Comment by ThomasMeng0515** [comment]
java屬於編譯式語言，編譯式語言會需要編譯器把程式碼編譯成機器讀得懂的機器碼，然後再使用機器碼去執行，在編譯的時候可以先對程式碼判斷有沒有錯誤，而且編譯式語言因為事先已經把程式編譯好所以執行速度會比直譯式語言(Ex: python)快。
更多相關資訊可以參考:
https://ithelp.ithome.com.tw/articles/10215893

**Comment by ghost** [comment]
不是腳本語言不代表程式執行的方式就不是由上至下一行一行執行
直譯、編譯、腳本語言
前兩者最主要的差別在於編譯與否，而腳本語言別與前兩者，他是負責解釋執行，並且其執行文件為文本

三者區別可參考：https://www.itread01.com/content/1547825905.html
